### DISCUSSION TITLE
Audio playback slower than normal?

### DISCUSSION THREAD

**Discussion by BB-check** [pschatzmann/arduino-audio-tools#409] [Q&A] [answered]
Hi,

I'm using the "streams-memory_raw-i2s" to get familiar with this library and play back audio straight from flash memory. I'm able to get the base example working with a MAX98357A board. However, whenever I play back any new audio files that I've generated other than the given example "StarWars30.h" file they seem to be playing somewhat slower than when I play back the audio file on the computer. 

Using audacity, I exported an audio file to a 16-bit signed PCM wav file matching the 22050Hz sample rate for the config object in the example. After I have the new 16 bit signed file, I open it in the "HxD" hex editor to save it as a C file. (https://mh-nexus.de/en/hxd/) I ensure that it's saved as an unsigned char array to match the original example. 

Would there be any other playback for the I2SConfig object that need to be changed? Or do I need to change some export/encoding settings in Audacity to ensure the correct playback speed for any given audio file?

**Comment by pschatzmann** [comment, marked as answer]
Double check the following settings, which have an impact on the speed:
- sample_rate
- channels

**Reply by BB-check** [reply]
Checking audacity again, I had to split the stereo track into two mono tracks then convert one of the mono tracks into a char array. Thanks for your help!
